lib/storage/metadata/oplog/PersistRingInterface.js

```diff
@@ -0,0 +1,287 @@
+                const size = stream._readableState.length;
+                this.spClient.put(
+                    stream,
+                    size,
+                    parameters,
+                    this.reqUid,
+                    (err, key) => {
+                        if (err) {
+                            this.logger.error(`sproxyd put ${filterName} error`, { err });
+                            return next(err);
+                        }
+                        const newData = {};
+                        newData.offset = offset;
+                        newData.key = key;
+                        return next(null, newData);
+                    });
+            },
+            /*
+             * Check if the Zookeeper node exists
+             */
+            (newData, next) => {
+                this.zkClient.exists(
+                    this.getZKPath(filterName),
+                    (err, stat) => {
+                        if (err) {
+                            this.logger.error(`exists ${filterName} error`, { err });
+                            return next(err);
+                        }
+                        let doesExist = false;
+                        if (stat) {
+                            doesExist = true;
+                        }
+                        return next(null, newData, doesExist);
+                    });
+            },
+            /*
+             * If the Zookeeper node exists read it.
+             * Else create it.
+             */
+            (newData, doesExist, next) => {
+                if (doesExist) {
+                    this.zkClient.getData(
+                        this.getZKPath(filterName),
+                        (err, _oldData) => {
+                            if (err) {
+                                this.logger.error(`getData ${filterName} error`, { err });
+                                return next(err);
+                            }
+                            const oldData = JSON.parse(_oldData);
+                            return next(null, newData, oldData);
+                        });
+                } else {
+                    this.zkClient.mkdirp(
+                        this.getZKPath(filterName),
+                        null,
+                        err => {
+                            if (err) {
+                                this.logger.error(`mkdirp ${filterName} error`, { err });
+                                return next(err);
+                            }
+                            return next(null, newData, null);
+                        });
+                }
+            },
+            /*
+             * Store the context in the Zookeeper node and delte the old sproxyd key.
+             */
+            (newData, oldData, next) => {
+                const _newData = JSON.stringify(newData);
+                this.zkClient.setData(
+                    this.getZKPath(filterName),
+                    new Buffer(_newData),
```

Comment:
```suggestion
                    Buffer.from(_newData),
```
`new Buffer(str)` has been deprecated